### PR TITLE
Doc: added comments for test functions in cli/commands/test

### LINKED ISSUE #57: Doc - Describe `list_test_entrypoints` using Rustdoc comments (three slashes: ///)
In an effort of having fully documented code, we need the way we list entry points in a Cairo repository to be documented.

In src/cli/commands/test, please proceed to comment above the `fn list_test_entrypoints` declaration in an equivalent way as `pub fn compile` is documented.

### LINKED ISSUE #60: Doc - Describe `test_single_entrypoint` using Rustdoc comments (three slashes: ///)
In an effort of having fully documented code, we need the way we run tests on a single Cairo entry point to be documented.

In src/cli/commands/test, please proceed to comment above the `pub(crate) fn test_single_entrypoint` declaration in an equivalent way as `pub fn compile` is documented.

### LINKED ISSUE #61: Doc - Describe `run_tests_for_one_file` using Rustdoc comments (three slashes: ///)
In an effort of having fully documented code, we need the way we run tests for a single file to be documented.

In src/cli/commands/test, please proceed to comment above the `fn run_tests_for_one_file` declaration in an equivalent way as `pub fn compile` is documented.

### PR DESCRIPTION
**Pull request by gdnathan**
Added documentation for most of the functions.
close #57
close #61
close #60 

Also added comments for other function/structure, not required by the issues

### CODE REVIEW

File: src/cli/commands/test/mod.rs [hunk 1]
```diff
@@ -146,10 +173,17 @@ fn setup_hint_processor() -> BuiltinHintProcessor {
 	hint_processor
 }
 
+///create a new ``Hooks`` object, with the followings hooks:
+/// - pre_step_instructions
```
Comment (Eikix) [inline src/cli/commands/test/mod.rs:177]: `pre_step_instruction`

File: src/cli/commands/test/mod.rs [hunk 2]
```diff
@@ -158,6 +192,22 @@ fn list_cairo_files(root: &PathBuf) -> Result<Vec<PathBuf>, ListCommandError> {
...
+/// compile a cairo file, returning a truple
+/// (path_to_original_code, path_to_compiled_code, entrypoints)
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```ignore
+/// let plain_path = PathBuf::from("path_to_your_program");
+/// let compiled_path = compile(&plain_path);
+/// let entypoints = list_test_entrypoints(compiled_path);
```
Comment (Eikix) [inline src/cli/commands/test/mod.rs:204]: `entrypoints`

**Review by Eikix — COMMENTED** [review]
nice PR! a couple of typos and will merge

File: src/cli/commands/test/mod.rs [hunk 3]
```diff
@@ -175,6 +225,26 @@ fn purge_hint_buffer(execution_uuid: &Uuid, output: &mut String) {
 	clear_buffer(execution_uuid);
 }
 
+/// Execute a single test.
+/// this function will tale a program and an entrypoint name, will search for this entrypoint and
```
Comment (Eikix) [inline src/cli/commands/test/mod.rs:229]: `tale` -> `take`

File: src/cli/commands/test/mod.rs [hunk 4]
```diff
@@ -37,6 +37,7 @@ use crate::{
 	hooks,
 };
 
+/// Enum containing the possible errors that you may encounter in the ``Test``module
```
Comment (Eikix) [inline src/cli/commands/test/mod.rs:40]: ``Test``module -> ``Test`` module

**Review by Eikix — APPROVED** [review]
lgtm

**Comment by 0x666c6f** [comment]
Hi @gdnathan ! You have a pending payment in Only Dust for this PR. Could you please contact me on Telegram so we can fix it ? My handle : @xflpt

**Comment by gdnathan** [comment]
Hey ! I contacted you as @gd_nathan on telegram.

**Pull request merged**